modules/validators.py

```diff
@@ -209,3 +211,133 @@ def fix(self, data: dict, issues: dict) -> dict:
+
+        elif "timestamp" in data and type(data["timestamp"]) == int:
+            issues.append(FeedbackValidatorIssue.CONVERT_UNIX_TO_DATETIME)
+
+        # A correct type is assigned
+        if (
+            "type" not in data
+            or type(data["type"]) != str
+            or data["type"] not in ["fact", "related", "stats", "other"]
+        ):
+            issues.append(FeedbackValidatorIssue.INVALID_TYPE)
+
+        #  A question exists
+        if (
+            "question" not in data
+            or type(data["question"]) != str
+            or no_content.match(data["question"])
+        ):
+            issues.append(FeedbackValidatorIssue.MISSING_QUESTION)
+
+        # An answer exists
+        if (
+            "answer" not in data
+            or type(data["answer"]) != str
+            or no_content.match(data["answer"])
+        ):
+            issues.append(FeedbackValidatorIssue.MISSING_ANSWER)
+
+        return issues
+
+    def fix(self, data: dict, issues: dict) -> dict:
+        """
+        Fixes feedback data. 
+        - Critical issues:
+            1. An invalid timestamp is present
+            2. An invalid type is present
+            3. No question or answer is provided
+
+        Parameters
+        ----------
+        - `data : dict` A feedback object {type: String, timestamp: datetime, question: String, answer: String}
+        - `issues: dict` lists of FeedbackValidatorIssues
+
+        Returns
+        -------
+        dict
+            A fixed copy of the data
+
+        Raises
+        ------
+        FeedbackValidatorError
+            when the issue with the feedback data is not fixable.
+
+        """
+
+        form = data.copy()
+
+        for issue in issues:
+            # fixes invalid timestamp (set to current datetime)
+            if issue == FeedbackValidatorIssue.INVALID_TIMESTAMP:
+                print("Inferred query timestamp on server")
+                form["timestamp"] = datetime.datetime.now()
+
+            # converts a valid unix timestamp to a Python Datetime object
+            if issue == FeedbackValidatorIssue.CONVERT_UNIX_TO_DATETIME:
+                form["timestamp"] = datetime.datetime.fromtimestamp(form["timestamp"])
+
+            # fixes invalid type (set to OTHER)
+            if issue == FeedbackValidatorIssue.INVALID_TYPE:
+                print(f"Changed query type from invalid form to 'other'")
+                form["type"] = "other"
+
+            # raises errors for missing answer or missing questions
+            if issue == FeedbackValidatorIssue.MISSING_ANSWER:
+                raise FeedbackValidatorError(
+                    self.error_messages[FeedbackValidatorIssue.MISSING_ANSWER]
+                )
+            if issue == FeedbackValidatorIssue.MISSING_QUESTION:
+                raise FeedbackValidatorError(
+                    self.error_messages[FeedbackValidatorIssue.MISSING_QUESTION]
+                )
```

Comment:
It might be cleaner for these to be if-else if-else rather than if statements.  What you have is not wrong though - feel free to ignore this if you disagree, I don't feel too strongly.

Reply:
Much cleaner with your suggestion 👍